presto-docs/src/main/sphinx/develop/release-process.rst

```diff
@@ -0,0 +1,252 @@
+   migration paths must be reverted:
+   
+   - **Must revert**: Breaking client-server protocol compatibility
+   - **Must revert**: Removing SQL syntax without deprecation warnings
+   - **Should revert**: Changing session/config property behavior without aliases
+   
+   Client libraries evolve slowly and many users cannot easily upgrade clients.
+
+**Backwards Incompatible SPI Changes Without Migration Path**
+   If a backwards incompatible change to the SPI is discovered that lacks the required migration path 
+   (e.g., no deprecation period, no reasonable defaults for new methods), the change should be reverted 
+   based on these criteria:
+   
+   - **Must revert**: Breaking documented SPI interfaces or core connectors (Hive, Iceberg, Delta, Kafka)
+   - **Should revert**: Breaking maintained connectors with active usage in the repository
+   - **Consider reverting**: Breaking experimental or rarely-used connectors (weigh maintenance burden)
+   
+   Consider both documented interfaces and public implementations in the Presto repository. 
+   Create a GitHub issue marked as "release blocker" to alert the release shepherd.
+
+**When NOT to Revert**
+
+**Minor Bugs With Simple Fixes**
+   Small issues that can be fixed quickly should be addressed with a forward fix rather than revert:
+   
+   - **Fix forward**: Typos, logging issues, minor UI problems
+   - **Fix forward**: Test failures that don't affect production code
+   - **Fix forward**: Documentation errors or missing documentation
+   
+   If the fix is simpler than the revert and can be completed before RC, prefer fixing forward.
+
+**Performance Trade-offs**
+   Performance changes with mixed impact require case-by-case evaluation based on community feedback:
+   
+   - **Evaluate carefully**: What's rare for one user may be critical for another
+   - **Consider configuration**: Can the optimization be made optional via session properties?
+   - **Gather data**: Solicit feedback from multiple organizations during RC testing
+   
+   If multiple users report significant regressions, consider reverting or adding a feature flag.
+   Always document performance changes and workarounds in release notes.
+
+**Proprietary or Hidden Infrastructure Dependencies**
+   Changes cannot be reverted based on impacts to proprietary infrastructure, private forks, or 
+   non-public connectors/plugins. All revert decisions must be justifiable using only publicly 
+   available code, documentation, and usage patterns visible in the open source project.
+
+**Feature Additions With Minor Issues**
+   New features that don't affect existing functionality should be fixed rather than reverted:
+   
+   - **Don't revert**: New functions with edge case bugs (fix the bugs)
+   - **Don't revert**: Optional features that can be disabled via configuration
+   - **Don't revert**: Experimental features clearly marked as such
+   
+   Consider adding feature flags if stability is a concern.
```

Comment:
I'd suggest a more strong statement that when feasible, new features should always have feature flags.